doc/pages/source/documentation/index.md

```diff
@@ -289,6 +296,25 @@ class AnAction < Dynflow::Action
 end
 ```
 
+{% info_block %}
+
+It's considered a good practice to use the just enough data for the
+input for the action to perform the job. That means not too much
+(such as using ActiveRecord's attributes), as it might have
+performance impact as well as causes issues when changing the
+attributes later.
+
+ On the other hand, the input should contain enough data to perform
+the job without needint to reaching to external sources. Therefore,
```

Comment:
"needint" -> "needing"
"without needing to reach to external..." or "without the need for reaching to external..." would be gramatically correct I think